FirebaseMessaging/Sources/FIRMessagingPubSub.m

```diff
@@ -126,11 +126,17 @@ - (void)updateSubscriptionWithToken:(NSString *)token
   };
 
   if ([[FIRInstanceID instanceID] tryToLoadValidCheckinInfo]) {
-    [self updateSubscriptionToTopic:topic
-                                    withToken:token
-                                      options:options
-                                 shouldDelete:shouldDelete
-                                      handler:completion];
+    FIRMessagingTopicAction action = FIRMessagingTopicActionSubscribe;
+    if (shouldDelete) {
+      action = FIRMessagingTopicActionUnsubscribe;
+    }
```

Comment:
_optional suggestion_: This can be made into a ternary:

```objc
FIRMessagingTopicAction action = shouldDelete ? 
    FIRMessagingTopicActionUnsubscribe : FIRMessagingTopicActionSubscribe
```

Reply:
Done.

Reply:
Done.